### PR TITLE
Feat/246 Add candle chart controls

### LINKED ISSUE #246: Add chart controls
Give a user a way to customise the candlestick chart.

## Acceptance criteria

- [ ] As a user I can choose from multiple intervals
- [ ] As a user I can choose from multiple chart types
- [ ] As a user I can choose from multiple studies
- [ ] As a user I can choose from multiple technical indicators

### PR DESCRIPTION
**Pull request by johnwalley**
Closes #246

### CODE REVIEW

**Comment by edd** [comment]
Dropdowns look a little clunky, I can't decide if it's just the padding on the right hand side for long labels, or the totally flat colours. This might be more @campbellssource than control feedback, but some padding work might be an easy win.

Screenshot 2022-04-20 at 17 22 17
Screenshot 2022-04-20 at 17 22 31

**Comment by edd** [comment]
The depth chart switch works, but the resulting depth chart is not presented usefully:

# Firefox (version numbers as above)
Note the scrollbars. For some reason this chart is much bigger than the available space

Screenshot 2022-04-20 at 17 26 46

# Safari
Much smaller than the available space.

Screenshot 2022-04-20 at 17 26 00

File: libs/chart/src/lib/trading-chart.tsx [hunk 1]
```diff
@@ -18,15 +48,140 @@ export const TradingChartContainer = ({
...
+                </DropdownMenuItemIndicator>
+                {overlayLabels[overlay]}
+              </DropdownMenuCheckboxItem>
+            ))}
+          </DropdownMenuContent>
+        </DropdownMenu>
+        <DropdownMenu>
+          <DropdownMenuTrigger>
+            <Button appendIconName="caret-down" variant="secondary">
+              Studies
```
Comment (edd) [inline libs/chart/src/lib/trading-chart.tsx:140]: @mattrussell36 / @bglownia do we have i18n stuff in yet or is this how we're going for now?

Comment (mattrussell36) [reply libs/chart/src/lib/trading-chart.tsx:140]: We have the `t` function. All it does is return the string you pass in, we are just using it to make it easier to swap out for a proper translation layer later on
```suggestion
{t('Studies')}
```

**Comment by edd** [comment]
👍 Merge time, I reckon. I have an issue that I'll put in a follow up, but it is not controls related. The improved dropdown layout is nice and the switching functionality is working great.

**Review by edd — APPROVED** [review]

**Comment by edd** [comment]
Probably follow-up as it's not control related:
Firefox/Safari/Brave, OSX: Light mode has a blue bar on the right hand side. Dark mode is fine:
Screenshot 2022-05-04 at 17 40 05
Screenshot 2022-05-04 at 17 38 16

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+>(({ className, inset = false, ...radioItemprops }, forwardedRef) => (
+  <DropdownMenuPrimitive.RadioItem
+    {...radioItemprops}
+    ref={forwardedRef}
+    className={classNames(getCheckboxItemClasses(inset), className)}
+  />
+));
+
+/**
+ * \Renders when the parent {@link DropdownMenuCheckboxItem} or {@link DropdownMenuRadioItem} is checked.
```
[the remark this answers is not in the document]
Comment (johnwalley) [reply libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx]: Fixed

**Pull request merged**